sdk/python/kfp/dsl/executor.py

```diff
@@ -191,7 +191,7 @@ def handle_single_return_value(self, output_name: str, annotation_type: Any,
             origin_type = getattr(annotation_type, '__origin__',
                                   None) or annotation_type
             # relax float-typed return to allow both int and float.
-            if origin_type == float:
+            if origin_type in {int, float}:
```

Comment:
This seems to be an orthogonal behavior change? And it seems this might be problematic to accept float when the int was expected?

Reply:
Yes. Thank you for catching that. This LOC was included in error.

I'll flag the correct solution to the problem this intended to solve in the next PR.